scm-ui/ui-components/src/forms/FileInput.tsx

```diff
@@ -66,18 +75,31 @@ const FileInput: FC<Props> = ({
   return (
     <div className={classNames("field", className)}>
       <LabelWithHelpIcon label={label} helpText={helpText} />
-      <div className="control">
-        <input
-          ref={ref}
-          name={name}
-          className={classNames("input", "p-1", className)}
-          type="file"
-          placeholder={placeholder}
-          disabled={disabled}
-          onChange={handleChange}
-          onBlur={handleBlur}
-          {...createAttributesForTesting(testId)}
-        />
+      <div className="file is-info has-name is-fullwidth">
+      <label className="file-label">
+          <input
+            ref={ref}
+            name={name}
+            className="file-input"
+            type="file"
+            placeholder={placeholder}
+            disabled={disabled}
+            onChange={handleChange}
+            onBlur={handleBlur}
+            {...createAttributesForTesting(testId)}
+          />
+          <span className="file-cta">
+            <span className="file-icon">
+              <i className="fas fa-arrow-circle-up" />
+            </span>
+            <span className="file-label has-text-weight-bold">{t("fileInput.label")}</span>
+          </span>
+          {file?.name ? (
+            <span className="file-name">{file?.name}</span>
+          ) : (
+            <span className="file-name has-text-weight-light has-text-grey-light">{t("fileInput.noFileChosen")}</span>
```

Comment:
Before we supported custom filename placeholder. This is useful to provide file restrictions like we did in the repository-avatar-plugin.

```suggestion
            <span className="file-name has-text-weight-light has-text-grey-light">{filenamePlaceholder || t("fileInput.noFileChosen")}</span>
```

Reply:
Only the FileUpload component supported this feature. FileInput did not. I can add it for future use 👍